### PR TITLE
Simplify package-info examples

### LINKED ISSUE #358: Standardize code snippets in package-info
Right now, the example in the Datastore's and Storage's package-info.java files explicitly sets the service options' project IDs but not the auth credentials.  The user will probably specify both or none, depending on whether they run the code from App/Compute Engine or not.

I think we should have two examples in each package info, one with both left unset and the other with both explicitly set.

### PR DESCRIPTION
**Pull request by ajkannan**
Fixes #358 

Instead of including two examples (one from App/Compute Engine, the other not), I added links to the "Specifying a Project ID" and "Authentication" sections of the README.

### CODE REVIEW

File: gcloud-java-datastore/src/main/java/com/google/gcloud/datastore/package-info.java [hunk 1]
```diff
@@ -47,6 +46,12 @@
  * }
  * } </pre>
  *
+ * <p>When using gcloud-java from outside of App/Compute Engine, you have to <a
+ * href="https://github.com/GoogleCloudPlatform/gcloud-java#specifying-a-project-id">specify a
+ * project ID</a> and
+ * <a href="https://github.com/GoogleCloudPlatform/gcloud-java#authentication">provide
```
Comment (mziccard) [inline gcloud-java-datastore/src/main/java/com/google/gcloud/datastore/package-info.java]: Maybe "setup authentication"?

File: gcloud-java-storage/src/main/java/com/google/gcloud/storage/package-info.java [hunk 2]
```diff
@@ -35,6 +34,11 @@
  *   channel.close();
  * }}</pre>
  *
+ * When using gcloud-java from outside of App/Compute Engine, you have to <a
+ * href="https://github.com/GoogleCloudPlatform/gcloud-java#specifying-a-project-id">specify a
+ * project ID</a> and
+ * <a href="https://github.com/GoogleCloudPlatform/gcloud-java#authentication">provide
```
Comment (mziccard) [inline gcloud-java-storage/src/main/java/com/google/gcloud/storage/package-info.java]: Also here, "setup authentication"?

File: gcloud-java-datastore/src/main/java/com/google/gcloud/datastore/package-info.java [hunk 3]
```diff
@@ -47,6 +46,12 @@
...
+ * authentication</a>.
+ *
```
Comment (mziccard) [inline gcloud-java-datastore/src/main/java/com/google/gcloud/datastore/package-info.java]: Nit, is this extra space needed?

File: gcloud-java-storage/src/main/java/com/google/gcloud/storage/package-info.java [hunk 4]
```diff
@@ -17,10 +17,9 @@
 /**
  * A client to Google Cloud Storage.
  *
- * <p>A simple usage example:
+ * <p>Here's a simple usage example for when running on App/Compute Engine:
```
Comment (mziccard) [inline gcloud-java-storage/src/main/java/com/google/gcloud/storage/package-info.java]: "for using gcloud-java from App/Compute Engine"

File: gcloud-java-datastore/src/main/java/com/google/gcloud/datastore/package-info.java (hunk 1 above)
Comment (aozarov) [inline gcloud-java-datastore/src/main/java/com/google/gcloud/datastore/package-info.java]: or "provide credentials"?

**Comment by ajkannan** [comment]
Suggestions incorporated

**Comment by mziccard** [comment]
Looks good to me

**Pull request merged**